Prefetching-Library/android_prefetching_lib/src/main/java/nl/vu/cs/s2group/nappa/PrefetchingLib.java

```diff
@@ -196,6 +164,109 @@ public static void init(Context context, int prefetchStrategyNum) {
 
     }
 
+    /**
+     * Query the database for the aggregate visit time and add an observer to the result
+     *
+     * @param activity A {@link ActivityNode} object contaning the activity name
+     */
+    private static void addVisitTimePerActivityObserver(@NotNull ActivityNode activity) {
+        if (activity.isAggregateVisitTimeInstantiated()) return;
+        if (prefetchingStrategyType != PrefetchingStrategyType.STRATEGY_GREEDY_VISIT_FREQUENCY_AND_TIME)
+            return;
+
+        Log.d(LOG_TAG, activity.activityName + " - Add aggregate visit time observer");
+
+        poolExecutor.execute(() -> {
+            LiveData<AggregateVisitTimeByActivity> liveData;
+
+            Object lastNSessionsObj = config.get(PrefetchingStrategyConfigKeys.LAST_N_SESSIONS);
+            int lastNSessions = lastNSessionsObj != null ?
+                    Integer.parseInt(lastNSessionsObj.toString()) : AbstractPrefetchingStrategy.DEFAULT_LAST_N_SESSIONS;
+
+            Object useSessionEntityObj = config.get(PrefetchingStrategyConfigKeys.USE_ALL_SESSIONS_AS_SOURCE_FOR_LAST_N_SESSIONS);
```

Comment:
I probably need to make variables/method with shorter names, look at the size of this